docs/rules/jsx-no-allocation-in-props.md

```diff
@@ -0,0 +1,34 @@
+# No Array or Object Allocation in JSX Props (react/jsx-no-allocation-in-props)
+
+An array literal `[]` or an object literal `{}` in a JSX prop will create a brand new object on every single render. This is bad for performance, as it will result in the garbage collector being invoked way more than is necessary. It may also cause unnecessary re-renders if a brand new object is passed as a prop to a component that uses reference equality check on the prop to determine if it should update.
+
+## Rule Details
+
+The following patterns are considered warnings:
+
+```jsx
+<Foo bar={[1, 2, 3]}></Foo>
+```
+```jsx
+<Foo bar={{ foo: 1 }}></Foo>
+```
+
+## Rule Options
+
+```js
+"react/jsx-no-allocation-in-props": [<enabled>, {
```

Comment:
array literals, and object literals, should probably have separate controls

Reply:
I actually couldn't come up with a use case where an array would be supplied as a prop to a DOM component, that probably is not a valid case?

Reply:
for DOM components sure, but i meant, i should be able to configure this rule to only warn on object literals, or to only warn on array literals.